qutebrowser/config/config.py

```diff
@@ -187,11 +188,25 @@ def get_reverse_bindings_for(self, mode: str) -> '_ReverseBindings':
     def get_command(self,
                     key: keyutils.KeySequence,
                     mode: str,
-                    default: bool = False) -> typing.Optional[str]:
+                    default: bool = False,
+                    pattern: urlmatch.UrlPattern = None) -> typing.Optional[str]:
         """Get the command for a given key (or None)."""
         self._validate(key, mode)
         if default:
+            if pattern is not None:
+                raise cmdutils.CommandError(
+                    "Cannot get defaults with a url pattern!")
             bindings = dict(val.bindings.default[mode])
+        elif pattern is not None:
+            # FIXME is there a way to avoid the to_py here?
+            opt = self._config.get_opt('bindings.commands')
```

Comment:
Just a heads up - this line is extremely costly (~40% of time spent when typing, to the point of causing lag when typing), but with #4707 this is only bad when there are per-domain bindings that could be matched. I might work on config cache for per-domain bindings just for this line.